internal/handlers/msg_service_router_test.go

```diff
@@ -406,7 +414,1712 @@ func TestMsgServiceAssessMsgFee(t *testing.T) {
+				},
+			},
+		},
+	)
+
+	txBytes, err := SignTxAndGetBytes(NewTestGasLimit(), sdk.NewCoins(sdk.NewInt64Coin("atom", 150), sdk.NewInt64Coin(msgfeestypes.NhashDenom, 1_190_500_000)), encCfg, priv.PubKey(), priv, *acct1, ctx.ChainID(), &rewardProgram)
+	require.NoError(t, err)
+	res := app.DeliverTx(abci.RequestDeliverTx{Tx: txBytes})
+	require.Equal(t, true, res.IsErr(), "should return error", res)
+}
+
+func TestRewardsProgramStart(t *testing.T) {
+	encCfg := simapp.MakeTestEncodingConfig()
+	priv, _, addr := testdata.KeyTestPubAddr()
+	acct1 := authtypes.NewBaseAccount(addr, priv.PubKey(), 0, 0)
+	acct1Balance := sdk.NewCoins(sdk.NewInt64Coin("hotdog", 1000), sdk.NewInt64Coin("atom", 1000), sdk.NewInt64Coin(msgfeestypes.NhashDenom, 1_190_500_000))
+	app := piosimapp.SetupWithGenesisAccounts([]authtypes.GenesisAccount{acct1}, banktypes.Balance{Address: addr.String(), Coins: acct1Balance})
+	ctx := app.BaseApp.NewContext(false, tmproto.Header{Time: time.Now()})
+	app.AccountKeeper.SetParams(ctx, authtypes.DefaultParams())
+	require.NoError(t, simapp.FundAccount(app.BankKeeper, ctx, acct1.GetAddress(), sdk.NewCoins(sdk.NewCoin(msgfeestypes.NhashDenom, sdk.NewInt(290500010)))))
+
+	rewardProgram := *rewardtypes.NewMsgCreateRewardProgramRequest(
+		"title",
+		"description",
+		acct1.Address,
+		sdk.NewInt64Coin("nhash", 1000),
+		sdk.NewInt64Coin("nhash", 100),
+		time.Now().Add(time.Duration(1)*time.Second),
+		9,
+		3,
+		3,
+		10,
+		[]rewardtypes.QualifyingAction{
+			{
+				Type: &rewardtypes.QualifyingAction_Transfer{
+					Transfer: &rewardtypes.ActionTransfer{
+						MinimumActions:          0,
+						MaximumActions:          10,
+						MinimumDelegationAmount: sdk.NewCoin("nhash", sdk.ZeroInt()),
+					},
+				},
+			},
+		},
+	)
+	app.BeginBlock(abci.RequestBeginBlock{Header: tmproto.Header{Height: 2, Time: time.Now().UTC()}})
+	txReward, err := SignTx(NewTestGasLimit(), sdk.NewCoins(sdk.NewInt64Coin("atom", 150), sdk.NewInt64Coin(msgfeestypes.NhashDenom, 1_190_500_000)), encCfg, priv.PubKey(), priv, *acct1, ctx.ChainID(), &rewardProgram)
+	require.NoError(t, err)
+	_, res, errFromDeliverTx := app.Deliver(encCfg.TxConfig.TxEncoder(), txReward)
+	require.NoError(t, errFromDeliverTx)
+	assert.Equal(t, true, len(res.GetEvents()) >= 1, "should have emitted an event.")
+	app.EndBlock(abci.RequestEndBlock{Height: 2})
+	app.Commit()
+
+	assert.Equal(t, 13, len(res.Events))
+	lookForRewardCreatedEvent, found := contains(res.Events, "reward_program_created")
+	assert.Equal(t, true, found, "event reward_program_created should exist")
+	assert.Equal(t, 1, len(*lookForRewardCreatedEvent), "event reward_program_created should exist")
+	lookForMessageEvent, foundMessage := contains(res.Events, "message")
+	assert.Equal(t, true, foundMessage, "event message should exist")
+	_, foundAttribute := containsAttribute(*lookForMessageEvent, "create_reward_program")
+	assert.Equal(t, true, foundAttribute, "event attribute create_reward_program should exist")
```

Comment:
Here again with the `contains` thing vs using the `.Contains` test. If one of these were to fail, the only context you would get is the msg and that something was true/false unexpectedly. You'd then have to spend extra time firing up the debugger in order to see what it was and what was missing.

Alternatively, create a function that takes in a `*testing.T` and is a `t.Helper()` that does the checking you want to be done, but make it output all of the thing being checked and what it's being checked for.

Reply:
Agreed, this can be simplified. Added two additional test helper functions.